dissect/ffs/ffs.py

```diff
@@ -221,7 +226,19 @@ def link(self):
         if not self.is_symlink():
             raise NotASymlinkError(f"{self!r} is not a symlink")
 
-        return self.open().read().decode("utf-8")
+        return self.open().read().decode(errors="surrogateescape")
+
+    @property
+    def link_inode(self):
+        if not self._link_inode:
+            # Relative lookups work because . and .. are actual directory entries
+            link = self.link
+            if link.startswith("/"):
+                relnode = None
+            else:
+                relnode = self.parent
+            self._link_inode = self.fs.get(self.link, relnode)
+        return self._link_inode
```

Comment:
```suggestion
    @cached_property
    def link_inode(self):
        # Relative lookups work because . and .. are actual directory entries
        link = self.link
        if link.startswith("/"):
            relnode = None
        else:
            relnode = self.parent
        return self.fs.get(self.link, relnode)
```